packages/terra-worklist-data-grid/src/subcomponents/Row.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import ThemeContext from 'terra-theme-context';
+import classNames from 'classnames/bind';
+import '../_elementPolyfill';
+import styles from './Row.module.scss';
+import RowSelectionCell from './RowSelectionCell';
+import Cell from './Cell';
+import columnShape from '../proptypes/columnShape';
+import rowShape from '../proptypes/rowShape';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The row's position in the Grid. This is zero based.
+   */
+  rowIndex: PropTypes.number,
+  /**
+   * String that specifies height of the row. Any valid CSS width value is accepted.
+  */
+  height: PropTypes.string,
+  /**
+   * The information about the Row.
+   */
+  row: rowShape.isRequired,
+  /**
+   * Boolean indicating whether or not the DataGrid should allow entire rows to be selectable. An additional column will be
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+  /**
+   * Callback function that will be called when a row is selected. Parameters: `onRowSelect(rowId)`
+   */
+  onRowSelect: PropTypes.func,
+  /**
+   * All columns currently displayed.
+   */
+  displayedColumns: PropTypes.arrayOf(columnShape),
+  /**
+   * Callback function that will be called when a cell in the row is selected.
+   */
+  onCellSelect: PropTypes.func,
+  /**
+   * Function to determine if a given Grid coordinates is the current tab stop on the Grid.
+   */
+  isTabStop: PropTypes.func,
+  /**
+   * Function used by the row to determine if a given cell in the row is currently selected.
+   */
+  isCellSelected: PropTypes.func,
```

Comment:
The WorklistDataGrid should be able to pass the selected cell id to the appropriate row instead of requiring a callback method.  When the selected cell is updated, it should trigger a re-render.  Maybe you are not using a useState hook?